mapiproxy/libmapiproxy/backends/openchangedb_mysql.c

```diff
@@ -3780,18 +3779,16 @@ enum MAPISTATUS openchangedb_mysql_initialize(TALLOC_CTX *mem_ctx,
 	oc_ctx->data = conn;
 	talloc_set_destructor(oc_ctx, openchangedb_mysql_destructor);
 	if (!table_exists(oc_ctx->data, "folders")) {
-		schema_dir = lpcfg_parm_string(lp_ctx, NULL, "openchangedb", "data");
-		schema_file = talloc_asprintf(mem_ctx, "%s/"SCHEMA_FILE,
-					      schema_dir ? schema_dir :
-					      openchangedb_data_dir());
-		DEBUG(3, ("Creating schema for openchangedb on mysql %s with %s\n",
-			  connection_string, schema_file));
-		schema_created = create_schema(oc_ctx->data, schema_file);
-		talloc_free(schema_file);
-
-		if (!schema_created) {
-			DEBUG(1, ("Failed openchangedb schema creation, "
-				  "last mysql error was: `%s`\n", mysql_error(conn)));
+		migration_cmd = talloc_asprintf(mem_ctx, "openchange_migrate "
+						"--openchangedb-uri=%s openchangedb", connection_string);
+		OPENCHANGE_RETVAL_IF(!migration_cmd, MAPI_E_NOT_ENOUGH_MEMORY, NULL);
+		DEBUG(3, ("Creating schema for openchangedb on mysql %s with this command %s\n",
+			  connection_string, migration_cmd));
+		schema_created_ret = system(migration_cmd);
```

Comment:
https://docs.python.org/2.5/ext/callingPython.html

We also have several references in mapistore-python branch where we call Python code from C